app/ota_client.py

```diff
@@ -257,91 +230,96 @@ def _header_str_to_dict(header_str):
     return header_dict
 
 
-def _save_update_ecuinfo(update_ecuinfo_yaml_file, update_ecu_info):
+def _save_update_ecuinfo(update_ecuinfo_yaml_file: Path, update_ecu_info: Path):
     """
     save update ecuinfo.yaml
     """
     output_file = update_ecuinfo_yaml_file
     logger.info(f"output_file: {output_file}")
-    with tempfile.NamedTemporaryFile("w", delete=False) as ftmp:
+    with tempfile.NamedTemporaryFile("w", delete=False, prefix=__name__) as ftmp:
         tmp_file_name = ftmp.name
-        with open(tmp_file_name, "w") as f:
-            f.write(yaml.dump(update_ecu_info))
-            f.flush()
+        ftmp.write(yaml.dump(update_ecu_info))
+
     shutil.move(tmp_file_name, output_file)
-    os.sync()
     return True
 
 
-class DirectoryInf:
+class _baseInf:
+    _base_pattern = re.compile(
+        r"(?P<mode>\d+),(?P<uid>\d+),(?P<gid>\d+),(?P<left_over>.*)"
+    )
+
+    def __init__(self, info: str):
+        match_res: re.Match = self._base_pattern.match(info.strip("\n"))
+        self.mode = int(match_res.group("mode"), 8)
+        self.uid = int(match_res.group("uid"))
+        self.gid = int(match_res.group("gid"))
+
+        self._left: str = match_res.group("left_over")
+
+
+class DirectoryInf(_baseInf):
     """
     Directory file information class
     """
 
     def __init__(self, info):
-        line = info.replace("\n", "")
-        info_list, last = _get_separated_strings(line, ",", 3)
-        self.mode = info_list[0]
-        self.uid = info_list[1]
-        self.gpid = info_list[2]
-        self.path = _decapsulate(line[last:])
+        super().__init__(info)
+        self.path = Path(self._left.strip("'"))
```

Comment:
Can this code parse the file name which contains `'`?
When the file name contains `'`, `'` is escaped with `'\''`.
https://tier4.atlassian.net/wiki/spaces/WEB/pages/1400443313/OTA+meta+data+file+format+specification

Reply:
I am confirming the documents to see what how to do it properly.

Reply:
Fix in 4d4eca4.
Now I use **.+** instead of **[^\']** to capture all possible characters.

Reply:
Fix in 17220ec.
Now leading \' and ending \' will only be stripped once, to allow path name like "\'\'\'\'\'abc\'\'\'\'" to be processed properly.
--------------------------
Also in 4d4eca4,
I use **.+** instead of **[^\']** to allow \' within the path.